components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,220 @@
+		checks[i] = *svcResp
+	}
+	serviceResponse.Checks = checks
+	sv.logger.Debug("The Passed value for the response: ", serviceResponse.Passed)
+	sv.logger.Debug("The Checks array for the response: ", serviceResponse.Checks)
+	return serviceResponse, nil
+}
+
+func (sv *SoftwareVersionService) checkCommandVersion(cmdName string) *models.Checks {
+	_, err := fiberutils.CheckPath(cmdName)
+	if err != nil {
+		sv.logger.Error("The errror which checking cammand file path: ", err)
+		return failureResponse(cmdName+AVAILABILITY, cmdName+" is not available", ENSURE+cmdName+" is available in $PATH on the node")
+	}
+	return successResponse(cmdName+AVAILABILITY, cmdName+" is available")
+}
+
+func (sv *SoftwareVersionService) checkOsVersion(osFilepath string) (*models.Checks, error) {
+	osVersions := map[string][]string{
+		"Red Hat":      {"7,8,9"},
+		"Ubuntu":       {"16.04.x", "18.04.x", "20.04.x", "22.04.x"},
+		"Centos":       {"7"},
+		"Amazon Linux": {"2"},
+		"SUSE Linux":   {"12"},
+		"Debian":       {"9", "10", "11", "12"},
+	}
+	checkResponse := &models.Checks{}
+	var osName, osVersion, err = getosutils.GetOsVersion(osFilepath)
+	if err != nil {
+		sv.logger.Error("Enable to get OS Version as the file on the path does not exit: ", err)
+		return failureResponse(LINUX_VERSION_CHECK, "Its not feasible to determine the Operating system version", "Please run automate on the supported platforms."), nil
+	}
+	sv.logger.Debug("Got the OS Version: ", osVersion)
+	sv.logger.Debug("Got the OS Name: ", osName)
+	for key := range osVersions {
+		if strings.Contains(strings.ToLower(osName), strings.ToLower(key)) {
+			correctVersion := sv.checkOs(osVersions, osVersion, key)
+			if correctVersion {
+				checkResponse = successResponse(LINUX_VERSION_CHECK, key+" version is "+osVersion)
+				break
+			}
+			checkResponse = failureResponse(LINUX_VERSION_CHECK, key+" version is not supported by automate", ENSURE+key+" correct version is installed on the node")
+			break
+		}
+		checkResponse = failureResponse(LINUX_VERSION_CHECK, osName+" version is not supported by automate", ENSURE+osName+" correct version is installed on the node")
+	}
+	return checkResponse, nil
+}
+
+func (sv *SoftwareVersionService) checkOs(osVersions map[string][]string, osVersion string, osName string) bool {
+	correctVersion := false
+	switch osName {
+	case UBUNTU:
+		split := strings.Split(osVersion, ".")
+		checkVersion := split[0] + "." + split[1]
+		re := regexp.MustCompile(checkVersion)
+
+		for _, str := range osVersions[osName] {
+			if re.MatchString(str) {
+				correctVersion = true
+				break
+			}
+		}
+	case RED_HAT:
+		checkVersion, _ := strconv.ParseFloat(osVersion, 64)
+		if checkVersion >= RED_HAT_SUPPORTED_VERSION {
+			correctVersion = true
+		}
+	case DEBIAN:
+		for _, str := range osVersions[osName] {
+			if str == osVersion {
+				correctVersion = true
+				break
+			}
+		}
+	default:
+		if osVersions[osName][0] == osVersion {
```

Comment:
Change the logic to check if the users' OS version is contained in `osVersions[osName] `